apicast/src/backend_client.lua

```diff
@@ -124,11 +124,23 @@ local function create_token_path(service_id)
   return format('/services/%s/oauth_access_tokens.xml', service_id)
 end
 
-local authorize_options = {
-  headers = {
-    ['3scale-options'] =  'rejection_reason_header=1'
-  }
-}
+-- Returns the authorize options that 3scale backend accepts. Those options
+-- are specified via headers. Right now there are 2:
+--  - rejection_reason_header: asks backend to return why a call is denied
+--    (limits exceeded, application key invalid, etc.)
+--  - no_nody: when enabled, backend will not return a response body. The
+--    body has many information like metrics, limits, etc. This information is
+--    parsed only when using oauth. By enabling this option will save some work
+--    to the 3scale backend and reduce network traffic.
+local function authorize_options(using_oauth)
```

Comment:
Wouldn't be nicer to pass `self` instead of `using_oauth` ? 

Because if the condition changes we would have to change only one place - this method.

Reply:
I see that it would also require changing the `authrep_path` method to expect self instead of just boolean.

Tough choice. Not so sure now. I guess it is fine as it is then!

Reply:
I like this option because it makes explicit that the only thing the method needs to know about is if we are running the oauth flow or not. Although as you said, we might need to change the params this method receives and the callers in the future.